### PR TITLE
Install mock by TypeLiteral and Qualifiers

### PR DESCRIPTION
**Pull request by gonciarz**

### CODE REVIEW

**Comment by geoand** [comment]
Thanks for this!

We'll need this to rebased onto `main` in order to get rid of the merge commit

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusMock.java [hunk 1]
```diff
@@ -56,4 +57,16 @@ public static <T> void installMockForType(T mock, Class<? super T> instance, Ann
         }
         MockSupport.installMock(CDI.current().select(instance, qualifiers).get(), mock);
     }
+
+    /**
+     * Installs a mock for a CDI normal scoped bean by typeLiteral and qualifiers
+     *
+     * @param mock          The mock object
+     * @param typeLiteral   TypeLiteral representing the required type qualifiers – the additional required qualifiers
```
Comment (mkouba) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusMock.java:65]: Pls remove the `qualifiers – the additional required qualifiers` part. The type literal has nothing to do with qualifiers ;-).

Comment (gonciarz) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusMock.java]: @mkouba done

Comment (mkouba) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusMock.java]: You misunderstood me - we should keep the `@param qualifiers` but remove the `qualifiers – the additional required qualifiers` part from the `@param typeLiteral`  line ;-)

Comment (gonciarz) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusMock.java]: @mkouba done again :)

**Comment by geoand** [comment]
@gonciarz would you like to rebase this onto main?

**Review by gsmet — APPROVED** [review]
I cleaned up the commits. It should be good to go now. Thanks!

**Pull request merged**